### ISSUE TITLE
Feature request: vcf output

### ISSUE THREAD

**Issue by lfaller** [mummer4/mummer#37]
This would be very helpful to visualize in IGV or Geneious!

Thank you,
~Lina

**Comment by gmarcais** [comment]
Can you give more details on what you envision. It seems that it is not a direct one-to-one transformation from an alignment file to a variant calling format.

**Comment by lfaller** [comment]
A lot of the folks in the lab at my institute use Geneious or IGV to look at alignments. These programs can read alignment files (bam, sam, etc) and they can also read a VCF file that contains SNP and Indel information. 

Unfortunately Mummer outputs SNP data in an uncommon format that is hard to overlay onto an alignment file.

My specific use case is as follows: I have two related bacterial genomes. I want to find out how similar they are. I used Mummer to align the two genomes and generated the SNP output. Now I would like to load one of the genomes into Geneious and add a track of the SNP information on top so I can see where the two genomes differ. I could easily do this with a VCF file.

Cheers,
~Lina

**Comment by lfaller** [comment]
Hello! Do you have any further thoughts on this?

Thanks!

**Comment by apredeus** [comment]
I agree - this would be an extremely useful feature. Of the few programs capable exporting the differences between the two genomes, most work quite .. unpredictably. It would be extremely useful to streamline finding SNPs and small indels, and annotate them with something like snpEff, to quickly assess if the variant is missense/nonsense/silent, identify frameshifts, etc. 

You are right of course that it's not a direct or easy transition. Large structural variations could be pretty hard to describe. But at least small indels and SNPs would be very useful.

**Comment by gmarcais** [comment]
I am OK with supporting an extra format, but the VCF is designed to report SNPs, not alignments. That is, to return a VCF files, some decision needs to be made on what or how to report, it is not just a matter of formatting the output differently. 

So I have 2 thoughts about that proposal:
* as it stands, this proposal is not precise enough. I understand your desire to integrate nucmer into your existing pipeline, but I would need more precise specifications of the desired behavior.
* there is already a tool called `show-snps`. Would the output of `show-snps`, if reformatted to the VCF format, meet your need?

**Comment by fritzsedlazeck** [comment]
Hey, sorry to jump into this discussion. It would be great to have an option in show-snps to output a VCF file. This would ease the usage of the results and also the comparison to e.g. mapping based methods. 

Do you know of any converter script?

Thanks
Fritz

**Comment by Ellis-Anderson** [comment]
Hey all,
I'm also here to advocate for `show-snps` to have a VCF output option. Trying to parse the current output into a vcf - particularly if you want to look at InDels - is not particularly easy. 

It would make sense to me that `show-snps` could produce an output that is rather standard across the field. 

Best,
Ellis

**Comment by apredeus** [comment]
Minimap2 now comes with a tool named paftools.js that allows reporting SNPs in VCF format, based on whole-genome comparison. Maybe nucmer developers can implement VCF output in a similar fashion, so that results would be more easily comparable between the two tools.

**Comment by shadiakiki1986** [comment]
For future-comers, this might be useful: https://www.biostars.org/p/395210/
It's a utility script that takes the `show-snps -T ...` output and converts it to vcf.
Also, check [here](https://github.com/MatteoSchiavinato/Utilities/issues/1) for a few examples.
